app/models/group.rb

```diff
@@ -19,6 +19,6 @@ class Group < ActiveRecord::Base
   has_many :members, through: :memberships, source: :user
 
   def user_can_join?(current_user)
-    !users_can_apply || current_user.membership_for(self)
+    users_can_apply && !current_user.membership_for(self)
```

Comment:
this reads a bit strangely to me. what do you think about something like: `users_can_apply? && member?(current_user)`

for this you might need to implement `Group#member?`. For me it reads better but maybe @tmichel will suggest something better